x/leverage/keeper/grpc_query.go

```diff
@@ -85,6 +85,10 @@ func (q Querier) Borrowed(
 		return &types.QueryBorrowedResponse{Borrowed: tokens}, nil
 	}
 
+	if !q.Keeper.IsAcceptedToken(ctx, req.Denom) {
+		return nil, status.Error(codes.InvalidArgument, "not accepted Token denom")
+	}
+
```

Comment:
Adjusted all queries that take a `denom` as argument, to return errors rather than zero when invalid assets are queried.

Reply:
Good idea.